src/convert.rs

```diff
@@ -669,6 +671,21 @@ impl TryFrom<tx_by_addr::TransactionError> for TransactionError {
                     ie,
                 ));
             }
+        } else if transaction_error.transaction_error == 30
```

Comment:
The code in validator which this file is copied from is:

        if let Some(transaction_details) = transaction_error.transaction_details {
            match transaction_error.transaction_error {
                30 => {
                    return Ok(TransactionError::DuplicateInstruction(
                        transaction_details.index as u8,
                    ));
                }
                31 => {
                    return Ok(TransactionError::InsufficientFundsForRent {
                        account_index: transaction_details.index as u8,
                    });
                }
                _ => {}
            }
        }